sdk/src/config.rs

```diff
@@ -298,16 +299,17 @@ fn load_inner<RwLock: lock_api::RawRwLock>(
                 base_path,
                 blueprint_id,
                 service_id,
+                protocol,
                 ..
             },
         ..
-    } = additional_config;
+    } = config;
 
     let logger = logger.unwrap_or_default();
 
     let data_dir = base_path;
```

Comment:
I'm still confused about the `data_dir_path`. Is it supposed to be the same as `keystore_url`? If I do `--base-path=":memory:"` then it won't be an actual *path*.

Reply:
Please address this @tbraun96

Reply:
Yeah, there was certainly redundancy. I have merged it all into a single field, using uri's instead.